Description: If an external server is used via client.conf and the DNS is inconsistent (ex: DNS gives "noname" for many IPs, reverse DNS gives one of these IPs for "noname") local PPDs can get polled for print queues instead of the PPDs of the external server
Bug: http://www.cups.org/str.php?L2763

Index: cups-1.6.0/cups/util.c
===================================================================
--- cups-1.6.0.orig/cups/util.c	2012-05-18 21:51:02.000000000 +0200
+++ cups-1.6.0/cups/util.c	2012-07-27 17:02:54.090195771 +0200
@@ -1079,11 +1079,13 @@
     http2 = http;
   else if ((http2 = httpConnectEncrypt(hostname, port,
                                        cupsEncryption())) == NULL)
-  {
-    DEBUG_puts("1cupsGetPPD3: Unable to connect to server");
+    if ((http2 = httpConnectEncrypt(http_hostname, http_port,
+				    cupsEncryption())) == NULL)
+    {
+      DEBUG_puts("1cupsGetPPD3: Unable to connect to server");
 
-    return (HTTP_SERVICE_UNAVAILABLE);
-  }
+      return (HTTP_SERVICE_UNAVAILABLE);
+    }
 
  /*
   * Get a temp file...
